### PR TITLE
[repo] Add-labels workflow fix

### PR DESCRIPTION
**Pull request by CodeBlanch**
Follow-up to #5601

## Changes

* Fix the `add-labels.yml` workflow so that the powershell properly handles issue body text containing newlines (see: https://github.com/open-telemetry/opentelemetry-dotnet/actions/runs/9033019307/job/24822395367)

## Merge requirement checklist

* [X] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)

### CODE REVIEW

**Review by Kielek — APPROVED** [review]

**Review by rajkumar-rangaraj — APPROVED** [review]

File: .github/workflows/add-labels.yml [hunk 1]
```diff
@@ -18,7 +18,12 @@ jobs:
       - name: Add labels for areas found in bug issue descriptions
         shell: pwsh
         run: |
-          $match = [regex]::Match('${{ github.event.issue.body }}', '^[#]+ Area\s*?(area:\w+)')
+          $body =
+          @"
+          ${{ github.event.issue.body }}
```
Comment (reyang) [inline .github/workflows/add-labels.yml:23]: Would this be a script injection concern? For example if the user put something like `"@` in the issue description? This bring a general concern regarding how are script injection are handled in other CI flows.

Comment (CodeBlanch) [reply .github/workflows/add-labels.yml]: Yikes ya you are right. I updated it so it doesn't inject anything into the script content.

File: build/scripts/add-labels.ps1 [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+)
+
+$match = [regex]::Match($issueBody, '^[#]+ Area\s*?(area:\w+)')
+if ($match.Success -eq $false)
+{
+    Return
+}
+
+gh issue edit $issueNumber `
+  --add-label $match.Groups[1].Value
```
Comment (reyang) [inline build/scripts/add-labels.ps1:13]:
```suggestion
gh issue edit $issueNumber --add-label $match.Groups[1].Value
```

**Review by reyang — APPROVED** [review]

**Comment by CodeBlanch** [comment]
Just thinking, should we make a dropdown for "signal" (tracing/logs/metrics/other?) on the bug template and also add a label for that?

**Comment by reyang** [comment]
> Just thinking, should we make a dropdown for "signal" (tracing/logs/metrics/other?) on the bug template and also add a label for that?

I guess this would be useful to the user for the feature request template, maybe start from there and see how things work, and if went well, come back to the bug template.

**Pull request merged**